x/crosschain/types/revert_options.go

```diff
@@ -76,6 +94,13 @@ func (r RevertOptions) GetEVMRevertAddress() (ethcommon.Address, bool) {
 	return addr, !crypto.IsEmptyAddress(addr)
 }
 
+// GetSOLRevertAddress returns the SOL revert address
+// if the revert address is not a valid address, it returns false
+func (r RevertOptions) GetSOLRevertAddress() (solana.PublicKey, bool) {
```

Comment:
We have the same pattern for other chains, but we should consider propagating this to the user. 
If the revert ends up on the sender address instead of the revert address, it might not be super clear for a dApp user. 
Wdyt? Maybe we should set this error into `error_revert,` which would be overwritten if the revert fails, but if the revert is successful, it would still be available